Update go-connections package

dmcgowan opened the pull request:
Changes windows message about system certificates from warn to info level. Ensures that server does not attempt to load an empty ca file.

fixes #30450
fixes https://github.com/docker/go-connections/issues/32

dmcgowan committed:
Update go-connections package

fixes #30450

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

cpuguy83 commented:
`pkg/listeners/listeners_unix.go:34: cannot use socketGroup (type string) as type int in argument to sockets.NewUnixSocket`

LK4D4 commented:
ping @dmcgowan

dmcgowan committed:
Convert socket group to int

Sockets interface has been updated to take in a the group
id as an integer rather than a string.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

dmcgowan commented:
Updated to convert the socket group to an integer.  That seems like the intent of https://github.com/docker/go-connections/pull/26.

cpuguy83 commented on pkg/listeners/listeners_unix.go:
```diff
@@ -31,7 +31,12 @@ func Init(proto, addr, socketGroup string, tlsConfig *tls.Config) ([]net.Listene
 	case "unix":
-		l, err := sockets.NewUnixSocket(addr, socketGroup)
+
+		gid, err := strconv.Atoi(socketGroup)
```
I think this really needs to be updated such that the function takes a gid as an integer, and then do group lookup in `cmd/dockerd/daemon.go`

cpuguy83 committed:
Do group lookup in listeners package

This used to be handled by go-connections, but now it only takes a group
ID (int).

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

cpuguy83 commented:
Pushed an update to this to use the proper group lookup and rebased.
Windows failure is some new thing that's failing everywhere (unrelated).

mlaventure reviewed (approved):
LGTM

ping @LK4D4

tiborvass commented:
LGTM

thaJeztah commented:
This was also in the 1.13 branch / 17.03 through https://github.com/docker/docker/pull/30730, so updated the milestone
